subsys/net/ip/igmp.c

```diff
@@ -139,7 +141,9 @@ static int send_igmp_report(struct net_if *iface,
 	}
 
 	for (i = 0; i < NET_IF_MAX_IPV4_MADDR; i++) {
-		if (!ipv4->mcast[i].is_used || !ipv4->mcast[i].is_joined) {
+		if (!ipv4->mcast[i].is_used || !ipv4->mcast[i].is_joined ||
+			net_ipv4_addr_cmp_raw((uint8_t *)&ipv4->mcast[i].address.in_addr,
```

Comment:
same here

Reply:
Thanks, I've incorporated this into my latest change.